### ISSUE TITLE
[2.2.6] Argument type compilation error

### ISSUE THREAD

**Issue by moElwan** [yireo/Yireo_ExtensionChecker#39]
## Preconditions
Magento 2.4.2-p1
php 7.4
Yireo_ExtensionChecker 2.2.6
Run `./bin/magento se:di:comp`

## Expected
Compilation finish without issues

## Result
Compilation error due to `Incompatible argument type` 
![image](https://user-images.githubusercontent.com/1895158/211317321-9de8aae5-fc35-4c94-93e4-5d55a440d8e5.png)

## Reason
It seems to be the phpdoc for this constructor https://github.com/yireo/Yireo_ExtensionChecker/blob/eddbd84761692c75c439c2618ee2dbe9bc802116/Console/Command/CreatePlantUmlDiagramCommand.php#L34

Will try to open a PR later today after testing the fix. Thank you

**Comment by jissereitsma** [comment]
Thanks for reporting this. I've tested this out myself under PHP 8.1 and PHP 7.4 but did not encounter any issue here. Also, the PHP docs of classes are actually not picked up upon by the ObjectManager class compilation (to my knowledge), so I fail to see why this would be an issue (unless you are using another third party addon that changes this behaviour). I've updated the docs anyway: https://github.com/yireo/Yireo_ExtensionChecker/commit/6ed30d1e3d1c9b425f520ed971656088b35eeec1 

Can you replicate this issue in a clean Magento 2.4.2 environment with only this extension and the Magento core?

**Comment by Maksold** [comment]
I also have an error on php 7.4 and magento 2.4.2-p2 on version 2.2.6.

```
The command "cd . && (/opt/homebrew/bin/php ./bin/magento setup:di:compile --no-ansi)" failed.

  Exit Code: 1(General error)

  Working directory: /xxx

  Output:
  ================
  Compilation was started.
  Errors during compilation:
  	Yireo\ExtensionChecker\Console\Command\CreatePlantUmlDiagramCommand
  		Incompatible argument type: Required type: string. Actual type: \Yireo\ExtensionChecker\PhpClass\ClassNameCollector; File:
  /xxx/vendor/yireo/magento2-extensionchecker/Console/Command/CreatePlantUmlDiagramCommand.php

  Total Errors Count: 1


  Error Output:
  ================
  %message% 0/9 [>---------------------------]   0% < 1 sec 102.0 MiB
  Proxies code generation... 0/9 [>-----------------------]   0% < 1 sec 102.0 MiB
  Proxies code generation... 1/9 [==>---------------------]  11% < 1 sec 106.0 MiB
  Repositories code generation... 1/9 [==>----------------]  11% < 1 sec 106.0 MiB
  Repositories code generation... 2/9 [====>--------------]  22% 6 secs 204.0 MiB
  Service data attributes generation... 2/9 [===>----------]  22% 6 secs 204.0 MiB
  Service data attributes generation... 3/9 [====>---------]  33% 6 secs 204.0 MiB
  Application code generator... 3/9 [====>---------]  33% 6 secs 204.0 MiB
  Application code generator... 4/9 [======>-------]  44% 12 secs 230.0 MiB
  Interceptors generation... 4/9 [======>-------]  44% 12 secs 230.0 MiB
  Interceptors generation... 5/9 [=======>------]  55% 38 secs 324.0 MiB
  Area configuration aggregation... 5/9 [=======>------]  55% 38 secs 324.0 MiB
  Area configuration aggregation... 6/9 [=========>----]  66% 48 secs 466.0 MiB
  Interception cache generation... 6/9 [=========>----]  66% 48 secs 466.0 MiB
  In Log.php line 92:

    Error during compilation


  setup:di:compile
  ```

**Comment by jissereitsma** [comment]
Thanks. Unfortunately, I'm still not able to duplicate this. Could you check for the version of the Symfony Console?
```bash
composer show symfony/console | grep -i versions
```

**Comment by Maksold** [comment]
`versions : * v4.4.49`

**Comment by jissereitsma** [comment]
This issue was automatically closed with #41 However, the signature of Symfony commands changed some time ago. I was surprised myself that this caused a DI error, while I could not duplicate the issue myself. And this might be breaking things with older versions of Magento. Anyway, I'm not a big fan of withholding fixes because of older versions, I'd rather move forwards. Many thanks @Maksold

**Comment by Maksold** [comment]
Thanks, @jissereitsma for the fast response!
